simulate/sam.go

```diff
@@ -86,9 +88,14 @@ func generateSamReadNoFlag(readName string, refName string, ref []dna.Base, star
 	}
 	copy(currSam.Seq[realSeqStartIdx-start:len(currSam.Seq)-(end-realSeqEndIdx)], ref[realSeqStartIdx:realSeqEndIdx])
 
+	// If ancient DNA, we simulate cytosine deamination error
+	if ancientErrorRate > 0 {
```

Comment:
Note to self that ancient error needs to happen before sequencing error for the simulation to make sense